### PR TITLE
Fix -jN for Python 3.7

### LINKED ISSUE #3145: Python 3.7 and multiple threads
SCons version: 3.0.1
Python version: 3.7
Python distribution: homebrew
SCons installed via pip
Platform: OS X 10.13.6
How to reproduce: run scons on any project

Description:
I recently got forcibly upgraded on my Mac to Python 3.7 via homebrew. After I went through and reinstalled all my packages, including SCons, I started seeing the following message when I run scons:

```
scons: warning: parallel builds are unsupported by this version of Python;
	ignoring -j or num_jobs option.
```

SCons seems to work fine with Python 3.7, it just emits this warning message. I did a little digging, and It seems that the `WITH_THREAD` config variable that SCons is using to test whether Python was configured with multiple threads is no longer supported, i.e., `sysconfig.get_config_var('WITH_THREAD')` used in Script/Main.py returns None even though multithreading *is* supported on the homebrew build.

I tried reverting to Python 3.6 through homebrew, but that doesn't seem to be supported, at least not directly based on what I could find via Google. I also tried using a build of Python 3.6 that I downloaded from python.org, but that build didn't seem to work very well. Scripts that worked fine with the homebrew build of Python 3.6 didn't seem to work correctly with the python.org build.

Based on what I'm reading on scons.org, it doesn't appear to me that SCons is supposed to directly support Python 3.7, yet; however, since I'm definitely running into fewer issues using Python 3.7 than I am with non-homebrew versions of Python 3.6, I'll continue using it. I'm just filing this issue as a heads-up for an issue that will probably arise when you begin testing on Python 3.7.

While I did not post this issue to the users list, I did look through all the archives back to the first of the year, and saw nothing obvious relating to Python 3.7. I would be nice if there were a search function for the archives. A google search pinned to site:scons.org didn't yield anything useful.

### PR DESCRIPTION
**Pull request by garyo**
Fix -jN for Python 3.7. This should fix #3145. Review appreciated!

### CODE REVIEW

File: src/engine/SCons/Script/Main.py [hunk 1]
```diff
@@ -1255,7 +1255,11 @@ def order(dependencies):
 
 
     is_pypy = platform.python_implementation() == 'PyPy'
-    python_has_threads = sysconfig.get_config_var('WITH_THREAD') or is_pypy
+    # As of 3.7, python removed support for threadless platforms.
+    # See https://www.python.org/dev/peps/pep-0011/
+    is_37_or_later = sys.version_info.major > 3 or \
```
Comment (serhiy-storchaka) [inline src/engine/SCons/Script/Main.py:1260]: You can write this test simpler:
```python
sys.version_info >= (3, 7)
```

Comment (bdbaddog) [reply src/engine/SCons/Script/Main.py:1260]: Ohh. I like that.

Comment (bdbaddog) [reply src/engine/SCons/Script/Main.py:1260]: @serhiy-storchaka  - I've merged this pull request. But I'm going to use your sys.version_info >= (3, 7).  Would you like some credit in the CHANGES.txt ? (If so, what name/identifier would you like used for such)?

Comment (serhiy-storchaka) [reply src/engine/SCons/Script/Main.py:1260]: It was a trivial suggestion, and this is an idiomatic code. It is not worth mentioning in CHANGES.txt.

Comment (bdbaddog) [reply src/engine/SCons/Script/Main.py:1260]: @serhiy-storchaka - OK.  Thanks!

**Pull request merged**